File: leetcode/1468-check-if-n-and-its-double-exist/solution.kt
class Solution {
    fun checkIfExist(arr: IntArray): Boolean {
        arr.sort()
        for ((i, x) in arr.withIndex()) {
            val j = arr.binarySearch(x*2)
            if(j >= 0 && j != i) {
                return true
            }
        }
        return false
    }
}

File: leetcode/1468-check-if-n-and-its-double-exist/solution.rs
impl Solution {
    pub fn check_if_exist(mut arr: Vec<i32>) -> bool {
        arr.sort();
        (0..arr.len()).any(|i| 
            arr.binary_search(&(2*arr[i]))
                .is_ok_and(|j| j!=i)
        )
    }
}
